src/day_1/ch3_q_learning/q_learning_problem.py

```diff
@@ -0,0 +1,148 @@
+"""Q-learning solution."""
+import time
+
+import gym
+import numpy as np
+from visualization import visualize_results
+
+
+class QLearning:
+    def __init__(
+        self,
+        env: gym.Env,
+        state_num: int,
+        action_num: int,
+        gamma: float = 0.99,
+        learning_rate: float = 0.01,
+        num_steps: int = 10000000,
+        epsilon: float = 0.8,
+    ) -> None:
+        self.env = env
+        self.action_num = action_num
+        self.gamma = gamma
+        self.num_steps = num_steps
+        self.learning_rate = learning_rate
+        self.epsilon = epsilon
+
+        # Initialize Q-function table
+        self.q_table = np.zeros((state_num, action_num))
+
+    def select_action(self, state: int) -> np.ndarray:
+        ###
+        # Problem 1:
+        # Please write the code that choose the action with highest Q-value at the current state
+        if np.random.rand() > self.epsilon:
+            action = None
+        # Please write the code that choose a random action with probability epsilon
+        else:
+            action = None
+        ###
```

Comment:
I think it would be better to let them do this whole part by themselves.
```suggestion
        ###
        # Problem 1:
        # Please write the code that chooses the action with the highest Q-value
        # at the current state with probability 1 - epsilon 
        # and a random action with probability epsilon
        if None:
            action = None
        else:
            action = None
        ###
```

Reply:
Done